### DISCUSSION TITLE
Can I pass body and multipart file into `RedirectResponse` in `post` method?

### DISCUSSION THREAD

**Discussion by tucan9389** [fastapi/fastapi#8639] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI

app = FastAPI()


@app.post("/")
def read_root(image_url: Optional[str] = Query('', alias="image_url"), file: UploadFile = File(Required, alias="file")):
    return RedirectResponse("/another")

@app.post("/another")
def read_another_root(image_url: Optional[str] = Query('', alias="image_url"), file: UploadFile = File(Required, alias="file")):
    return {"Hello": "World"}
```

### Description

When the server is requested at `/` in `post` method, I want to redirect the `/another` in `post` method with same body and multipart file.
How can I pass the original body and multipart file into the redirection endpoint?

### Operating System

Linux

### FastAPI Version

0.65.2

### Python Version

3.8.5

**Comment by ghandic** [comment]
What is the usecase? You could just add multiple paths for the same endpoint

```python
from fastapi import FastAPI

app = FastAPI()

@app.post("/")
@app.post("/another")
def read_another_root(image_url: Optional[str] = Query('', alias="image_url"), file: UploadFile = File(Required, alias="file")):
    return {"Hello": "World"}
```

**Comment by tucan9389** [comment]
@ghandic 
Thanks for your comment.

I have an old endpoint(`/`), and I want to add new endpoints that the features are already served in the old endpoint.
So I want to redirect the old endpoint(`/`) into separated endpoints with a parameter.

```
"/" -
    |--> "/a"        # redirect if   param=="a" in `/` endpoint request
    |--> "/b"        # redirect elif param=="b" in `/` endpoint request
    |--> "/c"        # redirect elif param=="c" in `/` endpoint request
```

**Comment by ghandic** [comment, marked as answer]
I don't think it is supported in the same way a GET is as the user would need to confirm the redirection 

User (Data) -> Server 1
Server 1 (Redirect to Server 2) -> User
User - who the heck is server 2... :) 

Since you are not redirecting to another server, you can just call the function directly

```python
from fastapi import FastAPI

app = FastAPI()


@app.post("/")
def read_another_root(demo: str):
    if demo.startswith("a"):
        return do_a(demo)
    return do_b(demo)


@app.post("/a")
def do_a(demo: str):
    return {"answer": "a"}


@app.post("/b")
def do_b(demo: str):
    return {"answer": "b"}

```

**Comment by tucan9389** [comment]
@ghandic 
Thanks for the additional comment with the kind example!
Actually, my code has used CBV[^1] with custom class. And I made the additional classes that confirm CBV decorator for new endpoints.
So in this case, I wasn't able to call the function directly like your codes. Then I needed to make the instance of the class and call the method of CBV. But I thought there would be more smart ways...

Do you have any idea when using the CBV class?

[^1]: https://fastapi-utils.davidmontague.xyz/user-guide/class-based-views/

**Comment by ghandic** [comment]
Best to ask @dmontagu on that 😃
